### PR TITLE
Inject `OrionClient` during block registration, save, and load

### PR DESCRIPTION
**Pull request by madkinsz**
Updates `Block.save`, `Block.load`, and `Block.register_types_and_schema` to use `@inject_client` which:
- Simplifies implementation and reduces nesting for readability
- Allows users to provide clients to `save` and `load` to avoid extra client startup
- Allows us to perform future optimizations for client reuse by updating `inject_client`

### Example

```python
async with get_client() as client:
    # Use client for other things
    ....
    Block.load("test", client=client)

   # Previously, `client` could not be passed and it would always create a new one
```

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a `fix`, `feature`, `enhancement`, `docs`, `collections`, or `maintenance` label categorizing the change.

### CODE REVIEW

**Comment by madkinsz** [comment]
This is failing with our good friend:

```

E                   sqlalchemy.exc.DBAPIError: (sqlalchemy.dialects.postgresql.asyncpg.Error) <class 'asyncpg.exceptions.InFailedSQLTransactionError'>: current transaction is aborted, commands ignored until end of transaction block
E                   [SQL: SELECT block_type.id, block_type.created, block_type.updated, block_type.name, block_type.slug, block_type.logo_url, block_type.documentation_url, block_type.description, block_type.code_example, block_type.is_protected 
E                   FROM block_type 
E                   WHERE block_type.slug = %s]
E                   [parameters: ('x',)]
E                   (Background on this error at: https://sqlalche.me/e/14/dbapi)
```

During: `tests/orion/database/test_migrations.py::test_orion_full_migration_works_with_data_in_db`

I believe this issue will be resolved in #6858 and this pull request may need to wait until that is merged.

**Review by desertaxle — APPROVED** [review]
LGTM!

**Comment by madkinsz** [comment]
Some service tests failing with unrelated conda image cache export issues. Guidry and I are addressing separately.

**Pull request merged**
